stdlib/public/core/Codable.swift

```diff
@@ -5510,23 +5510,101 @@ internal struct _DictionaryCodingKey: CodingKey {
+/// custom type and a `CodingKey` type.
+///
+/// Conforming a type to `CodingKeyRepresentable` lets you opt in to encoding and
+/// decoding `Dictionary` values keyed by the conforming type to and from a keyed
+/// container, rather than encoding and decoding the dictionary as an unkeyed container
+/// of alternating key-value pairs.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public protocol CodingKeyRepresentable {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  var codingKey: CodingKey { get }
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  init?<T: CodingKey>(codingKey: T)
+}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension RawRepresentable where Self: CodingKeyRepresentable, RawValue == String {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public var codingKey: CodingKey {
+    _DictionaryCodingKey(stringValue: rawValue)
+  }
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public init?<T: CodingKey>(codingKey: T) {
+    self.init(rawValue: codingKey.stringValue)
+  }
+}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension RawRepresentable where Self: CodingKeyRepresentable, RawValue == Int {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public var codingKey: CodingKey {
+    _DictionaryCodingKey(intValue: rawValue)
+  }
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public init?<T: CodingKey>(codingKey: T) {
+    if let intValue = codingKey.intValue {
+      self.init(rawValue: intValue)
+    } else {
+      return nil
+    }
+  }
+}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension Int: CodingKeyRepresentable {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public var codingKey: CodingKey {
+    _DictionaryCodingKey(intValue: self)
+  }
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public init?<T: CodingKey>(codingKey: T) {
+    if let intValue = codingKey.intValue {
+      self = intValue
+    } else {
+      return nil
+    }
+  }
+}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension String: CodingKeyRepresentable {
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public var codingKey: CodingKey {
+    _DictionaryCodingKey(stringValue: self)
+  }
+  @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+  public init?<T: CodingKey>(codingKey: T) {
+    self = codingKey.stringValue
+  }
 }
 
 extension Dictionary: Encodable where Key: Encodable, Value: Encodable {
   /// Encodes the contents of this dictionary into the given encoder.
   ///
-  /// If the dictionary uses `String` or `Int` keys, the contents are encoded
-  /// in a keyed container. Otherwise, the contents are encoded as alternating
-  /// key-value pairs in an unkeyed container.
+  /// If the dictionary uses keys that are `String`, `Int` or a type conforming to
```

Comment:
Sorry, missed this on the first pass:

```suggestion
  /// If the dictionary uses keys that are `String`, `Int`, or a type conforming to
```

Apple style includes the comma before "and" and "or" — known as a serial comma.  This follows style guides like Chicago Manual of Style that are generally used more for books, and contrasts with newspaper style guides that generally omit it.

Reply:
Thanks! Nice to learn about serial commas! :-)